airflow/api/common/mark_tasks.py

```diff
@@ -159,6 +159,7 @@ def set_state(
             tis_altered += session.scalars(qry_sub_dag.with_for_update()).all()
         for task_instance in tis_altered:
             task_instance.set_state(state, session=session)
+            task_instance.dag_run.activate_scheduling(session)
```

Comment:
I no longer need this. I will verify

Reply:
It turns out that we still need it. When a task is in up_for_retry state, and we mark it as success/failed, the scheduler process executor events don't activate the dagrun. This might be a different bug. It might be better not to take chances